### PR TITLE
Store vcs revision in metadata

### PR DESCRIPTION
**Pull request by yglukhov**
Basically subj. The reason behind this feature request is that we need some way to track our package revisions, that change in git many times a day, and manually updating version numbers in nimble files is cumbersome and error prone. For now I just want to test this feature with no public api exposed to see how it goes.

### CODE REVIEW

File: src/nimble.nim [hunk 1]
```diff
@@ -326,11 +326,14 @@ proc buildFromDir(pkgInfo: PackageInfo, paths: seq[string], forRelease: bool) =
...
+proc saveNimbleMeta(pkgDestDir, url, vcsRevision: string, filesInstalled: HashSet[string]) =
   var nimblemeta = %{"url": %url}
-  nimblemeta["files"] = newJArray()
+  if not vcsRevision.isNil:
+    nimblemeta["vcsRevision"] = %vcsRevision
+  let files = newJArray()
+  nimblemeta["files"] = files
   for file in filesInstalled:
-    nimblemeta["files"].add(%changeRoot(pkgDestDir, "", file))
+    files.add(%changeRoot(pkgDestDir, "", file))
```
Comment (dom96) [inline src/nimble.nim]: Why this change? Just didn't want to write out ``nimblemeta["files"]``?

Comment (yglukhov) [inline src/nimble.nim]: Yep, also a performance improvement! =)
